### PR TITLE
Back-off exponentially when retrying failed SCS calls

### PR DESCRIPTION
**Pull request by keirog**
Relies on [but-you-promised v1.1.0](https://github.com/keirog/but-you-promised/releases/tag/v1.1.0)

 🐿 v2.10.2

### CODE REVIEW

**Comment by AbdullahAli** [comment]
Our current version of `istanbul` does not recognise `async/await` syntax, which causes warnings in the test suite:

```
Failed to parse file: /Users/abdullahali/ft/email-send-api/app/services/consentService/fetcher.js
Transformation error; return original code
{ Error: Line 23: Unexpected identifier
    at constructError (/Users/abdullahali/ft/email-send-api/node_modules/istanbul/node_modules/esprima/esprima.js:2407:21)
    at createError (/Users/abdullahali/ft/email-send-api/node_modules/istanbul/node_modules/esprima/esprima.js:2426:17)
    at unexpectedTokenError (/Users/abdullahali/ft/email-send-api/node_modules/istanbul/node_modules/esprima/esprima.js:2500:13)
...
```




A workaround is to upgrade: https://github.com/gotwarlost/istanbul/issues/733#issuecomment-282227197

File: app/services/consentService/fetcher.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ const defaultFetchOptions = {
 
 const deepClone = obj => JSON.parse(JSON.stringify(obj));
 const buildFetchOptions = options => Object.assign(deepClone(defaultFetchOptions), options);
+const exponentialBackoff = ({ seedDelayInMs }) => {
+	return attemptsSoFar => (attemptsSoFar * attemptsSoFar) * seedDelayInMs;
```
[the remark this answers is not in the document]
Comment (keirog) [reply app/services/consentService/fetcher.js]: It’s creating a closure as a callback function, so once exponentialBackoff is initiated (kind-of) with something like `exponentialBackoff({ seedDelayInMs: 1000 })`, we pass that closure to but-you-promised, which then calls it and passes in an `attemptsSoFar` number on each attempt.

Lemme change it a bit; see if you find it more readable

**Comment by keirog** [comment]
Ooh now’s probably a good time to just move over to [nyc](https://github.com/istanbuljs/nyc) as [istanbul is deprecated](https://github.com/istanbuljs/istanbuljs#where-should-i-start)

File: app/services/consentService/fetcher.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AbdullahAli) [reply app/services/consentService/fetcher.js]: I think we can go ahead with it as is for now (to validate if it makes a difference) - we can revisit the refactor when we refactor other parts of the project

[the remark this answers is not in the document]
Comment (keirog) [reply app/services/consentService/fetcher.js]: 🙇

**Comment by AbdullahAli** [comment]
me_gusta_mucho_by_megustamuchoplz-d416uqk

**Pull request merged**